### PR TITLE
Implemented common cli app config in clix package

### PR DESCRIPTION
**Pull request by michalkurzeja**

### CODE REVIEW

File: clix/flags.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+var Defaults = defaults{
+	HTTPPort:  80,
+	LogFormat: "json",
+	LogLevel:  "info",
+
+	ProfilerPort: 8081,
+}
+
+var flags = []cli.Flag{
+	cli.IntFlag{
```
Comment (nrwiersma) [inline clix/flags.go]: Split out to `ServerFlags`

File: clix/flags.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+var Defaults = defaults{
+	HTTPPort:  80,
+	LogFormat: "json",
+	LogLevel:  "info",
+
+	ProfilerPort: 8081,
+}
+
+var flags = []cli.Flag{
```
Comment (nrwiersma) [inline clix/flags.go]: Remane to `CommonFlags`

File: clix/flags.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	},
+	cli.IntFlag{
+		Name:   FlagProfilerPort,
+		Value:  Defaults.ProfilerPort,
+		Usage:  "Port for the profiler to listen on.",
+		EnvVar: "PROFILER_PORT",
+	},
+}
+
+func Flags() []cli.Flag {
```
Comment (nrwiersma) [inline clix/flags.go]: This isnt really needed.

File: clix/flags.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	return flags
+}
+
+func ProfilerFlags() []cli.Flag {
+	return profilerFlags
+}
+
+func splitTags(slice []string, sep string) ([]interface{}, error) {
```
Comment (nrwiersma) [inline clix/flags.go]: Expose this so we can reuse it.

File: clix/flags.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+)
+
+const (
+	FlagPort = "http-port"
+
+	FlagLogFormat = "log-format"
+	FlagLogLevel  = "log-level"
+	FlagLogTags   = "log-tags"
+
+	FlagStatsAddr   = "stats-addr"
```
Comment (nrwiersma) [inline clix/flags.go]: This isnt an address, its a DSN.

File: clix/flags.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/urfave/cli"
+)
+
+const (
+	FlagPort = "http-port"
```
Comment (nrwiersma) [inline clix/flags.go]: make this `port` which is more standard.

File: clix/log.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+	return logger, nil
+}
+
+func newLogFormat(c Ctx) (log15.Format, error) {
+	format := c.String(FlagLogFormat)
+	switch format {
+	case "terminal":
+		return log15.TerminalFormat(), nil
+	case "json":
```
Comment (nrwiersma) [inline clix/log.go]: Please add an option for `LogFmt`

File: cache/redis_internal_test.go [hunk 8]
```diff
@@ -3,9 +3,10 @@ package cache
 import (
 	"testing"
 
+	"time"
```
Comment (nrwiersma) [inline cache/redis_internal_test.go]: Group the std lib imports

File: clix/context.go [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+	ctxContext
+}
+
+func NewContext(c *cli.Context, opts ...CtxOptionFunc) (*Context, error) {
+	ctx := &Context{
+		Context:    c,
+		ctxContext: context.Background(),
+	}
+
+	if len(opts) == 0 {
```
Comment (nrwiersma) [inline clix/context.go]: If you provide a log and not stats, then you get no defaults?

File: clix/context.go [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+func (c *Context) Close() error {
+	s, ok := stats.FromContext(c)
+	if ok {
+		return s.Close()
+	}
+
+	return nil
+}
+
+func Logger(l log.Logger) CtxOptionFunc {
```
Comment (nrwiersma) [inline clix/context.go]: group with the CtxOptionFunc

File: clix/context_test.go [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/msales/pkg/clix"
+	"github.com/msales/pkg/log"
+	"github.com/msales/pkg/stats"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
```
Comment (nrwiersma) [inline clix/context_test.go]: remove the empty line

File: clix/flags.go (hunk 1 above)
Comment (michalkurzeja) [reply clix/flags.go]: Done

File: clix/flags.go (hunk 2 above)
Comment (michalkurzeja) [reply clix/flags.go]: Done

File: clix/flags.go (hunk 4 above)
Comment (michalkurzeja) [reply clix/flags.go]: Good idea, done.

File: clix/flags.go (hunk 5 above)
Comment (michalkurzeja) [reply clix/flags.go]: Done

**Pull request merged**